src/core/js/a11y.js

```diff
@@ -59,353 +54,389 @@ define([
+  }
 
-    },
-
-    _browserFocus: new BrowserFocus(),
-    _keyboardFocusOutline: new KeyboardFocusOutline(),
-    _wrapFocus: new WrapFocus(),
-    _popup: new Popup(),
-    _scroll: new Scroll(),
-
-    log: new Log(),
-
-    initialize: function() {
-      this._removeLegacyElements();
-      this.listenToOnce(Adapt, {
-        'app:dataLoaded': this._onDataLoaded,
-        'navigationView:postRender': this._removeLegacyElements
-      }, this);
-      Adapt.on('device:changed', this._setupNoSelect);
-      this.listenTo(Adapt, {
-        'router:location': this._onNavigationStart,
-        'contentObjectView:ready router:plugin': this._onNavigationEnd
-      });
-    },
-
-    _onDataLoaded: function() {
-      this.config = Adapt.config.get('_accessibility');
-      this.config._isActive = false;
-      this.config._options = _.defaults(this.config._options || {}, this.defaults);
-      Adapt.offlineStorage.set('a11y', false);
-      this.$html.toggleClass('has-accessibility', this.isEnabled());
-      this._setupNoSelect();
-      this._addFocuserDiv();
-      if (this._isReady) {
-        return;
+  _setupNoSelect() {
+    if (!this.config || !this.config._disableTextSelectOnClasses) {
+      return;
+    }
+    const classes = this.config._disableTextSelectOnClasses.split(' ');
+    let isMatch = false;
+    for (let i = 0, item; (item = classes[i++]);) {
+      if (this.$html.is(item)) {
+        isMatch = true;
+        break;
       }
-      this._isReady = true;
-      Adapt.trigger('accessibility:ready');
-    },
+    }
+    this.$html.toggleClass('u-no-select', isMatch);
+  }
 
-    _setupNoSelect: function() {
-      if (!this.config || !this.config._disableTextSelectOnClasses) {
-        return;
-      }
-      var classes = this.config._disableTextSelectOnClasses.split(' ');
-      var isMatch = false;
-      for (var i = 0, item; (item = classes[i++]);) {
-        if (this.$html.is(item)) {
-          isMatch = true;
-          break;
-        }
+  _addFocuserDiv() {
+    if ($('#a11y-focuser').length) {
+      return;
+    }
+    $('body').append($('<div id="a11y-focuser" class="a11y-ignore" tabindex="-1" role="presentation">&nbsp;</div>'));
+  }
+
+  _removeLegacyElements() {
+    const $legacyElements = $('body').children('#accessibility-toggle, #accessibility-instructions');
+    const $navigationElements = $('.nav').find('#accessibility-toggle, #accessibility-instructions');
+    if (!$legacyElements.length && !$navigationElements.length) {
+      return;
+    }
+    Adapt.log.warn('REMOVED: #accessibility-toggle and #accessibility-instructions have been removed. Please remove them from all of your .html files.');
+    $legacyElements.remove();
+    $navigationElements.remove();
+  }
+
+  _onNavigationStart() {
+    if (!this.isEnabled()) {
+      return;
+    }
+    // Stop document reading
+    _.defer(function() {
+      Adapt.a11y.toggleHidden('.contentobject', true);
+    });
```

Comment:
We could probably do with setting up a bit in the style guide about use of arrow functions. I've seen a few that argue that you shouldn't use a one-liner arrow function in this situation since:
* there's no need to bind the scope of `this`
* there's no implicit `return`

Reply:
Fair dos! Personally, I favour readability over anything else – find it quicker to scan without the keyword/brackets etc., but is just an opinion.

Reply:
I don't know about those points. It sounds like a really weird way of rationalising the 'I really don't like the look of this new syntax' stance.

* there's no need to bind the scope of this

Every function has a call time `this` binding. It's the difference between:
```js
var spartacus = function() { 
  console.log(this.name || "he's a very naughty boy!"); 
};
var a = { name: "i am spartacus", spartacus };
var b = { name: "no! i am spartacus", spartacus };
a.spartacus();
b.spartacus();
spartacus();
```
Where the same function has different `this` context based upon the implied calling context,

* there's no implicit return
All functions have an implied `return undefined;`, whether we use the return value or not.